source/Subsystem/Yellowstone/Texture.cpp

```diff
@@ -0,0 +1,47 @@
+#include "Texture.h"
+#include "GLError.h"
+
+#include <glad/glad.h>
+#define STB_IMAGE_IMPLEMENTATION
+#include <stb_image.h>
+#include <fstream>
+#include <sstream>
+
+Texture::Texture(const std::string& aPath)
+	: myID(0)
+	, myWidth(0)
+	, myHeight(0)
+{
+	unsigned char* image = stbi_load(aPath.c_str(), &myWidth, &myHeight, &myComponents, STBI_default);
```

Comment:
Similar to what I had mentioned in the Model class, I would not recommend putting this type of function in a class that is going to be used a lot across the program. Having the loading function centralized and in a shared class/manager would save some memory usage if there a lot of texture objects present.

Reply:
I agree. This code is ported from my OpenGL sandbox, so it doesn't have the system design we want yet. I'll look into it.